### ISSUE TITLE
Remove IIFE in modules.commonjs plugin

### ISSUE THREAD

**Issue by eventualbuddha** [esnext/esnext#81]
If the entire program is a single IIFE statement whose callee is a function expression with no `id` and a call with no arguments, we can simply strip it since the closure created by it is redundant in a module setting anyway.

```js
(function() {
  // stuff
})();
```

becomes:

```js
// stuff
```

**Comment by eventualbuddha** [comment]
Also:

```js
(function() {
  // stuff
}).call(this);

// ↑ becomes ↓

// stuff
```

**Comment by eventualbuddha** [comment]
Released in v1.11.0.
